index.d.ts

```diff
@@ -16,20 +18,20 @@ export type Mapper<ValueType, KeyType, MappedValueType> = (
 export default function pProps<
 	KeyType extends unknown,
 	ValueType extends unknown,
-	MappedValueType = ValueType
+	MappedValueType = PromiseResult<ValueType>
```

Comment:
The `MappedValueType` can never be a `Promise(Like)`, all promises are resolved before being put into the resulting `Map`.

Reply:
`PromiseResult<Promise<T>>` extracts `T`, so `MappedValueType` would be a real resolved value from `ValueType`

Reply:
Right, sorry.